### PR TITLE
[CALCITE-6559] Query with measure that applies AVG to SMALLINT throws AssertionError "Cannot add expression of different type to set"

### PR DESCRIPTION
**Pull request by julianhyde**

### CODE REVIEW

File: core/src/test/resources/sql/measure.iq [hunk 1]
```diff
@@ -179,6 +179,53 @@ from empm;
...
+| CLERK     |      7769 |
+| MANAGER   |      7682 |
+| PRESIDENT |      7839 |
+| SALESMAN  |      7629 |
++-----------+-----------+
+(5 rows)
+
+!ok
+
+!if (false) {
```
Comment (mihaibudiu) [inline core/src/test/resources/sql/measure.iq:205]: Is there an issue filed for this other problem?

**Review by mihaibudiu — APPROVED** [review]

**Comment by julianhyde** [comment]
@amarkowitz, PTAL

File: core/src/test/resources/sql/measure.iq [hunk 2]
```diff
@@ -179,6 +179,53 @@ from empm;
 
 !ok
 
+# [CALCITE-6559] LogicalCorrelate Type mismatch
+# Occurs when AVG measure is applied to SMALLINT or TINYINT column.
+# A CAST is added internally to the result of SUM / COUNT, but this cast
```
Comment (amarkowitz) [inline core/src/test/resources/sql/measure.iq:184]: It's difficult to tell from this comment whether this was the cause of the bug or whether this is the new expectation after fixing.  Have you considered moving this comment to CALCITE-6559 instead?

File: core/src/test/resources/sql/measure.iq (hunk 1 above)
Comment (julianhyde) [reply core/src/test/resources/sql/measure.iq:205]: I've logged https://issues.apache.org/jira/browse/CALCITE-6564

**Pull request closed without merging**